src/app/payment/components/checkout/SelectSeatsComponent.tsx

```diff
@@ -1,66 +1,99 @@
+    setShowButton(false);
+  };
+
   return (
-    <div
-      role="menuitem"
-      tabIndex={0}
-      onKeyDown={(e) => {
-        e.stopPropagation();
-      }}
-      className="flex w-max flex-row items-center rounded-lg border"
-    >
-      <button
-        disabled={disableMinusButton}
-        onClick={(e) => {
-          e.preventDefault();
-          onUsersChange(seats - 1);
-        }}
-        className="flex h-full flex-col items-center justify-center rounded-l-lg px-4 hover:bg-gray-10"
-      >
-        -
-      </button>
-      <SeparatorVertical />
-      <input
-        type="text"
-        className="flex w-10 items-center justify-center !rounded-none border-0 text-center !outline-none !ring-0"
-        value={seats}
-        onChange={(e) => {
-          e.preventDefault();
-          onUsersChange(Number(e.target.value));
-        }}
+    <div className="flex flex-row gap-4">
+      <div
+        role="menuitem"
+        tabIndex={0}
         onKeyDown={(e) => {
           e.stopPropagation();
         }}
-        onBlur={(e) => {
-          e.preventDefault();
-
-          onUsersChange(seats);
-        }}
-      />
-      <SeparatorVertical />
-      <button
-        disabled={disablePlusButton}
-        onClick={(e) => {
-          e.preventDefault();
-          onUsersChange(seats + 1);
-        }}
-        className="flex h-full flex-col items-center justify-center rounded-r-lg px-4 hover:bg-gray-10"
+        className="flex w-max flex-row items-center rounded-lg border"
       >
-        +
-      </button>
+        <button
+          disabled={disableMinusButton}
+          onClick={(e) => {
+            e.preventDefault();
+            onSeatsChange(seats - 1);
+            setTotalUsers(seats - 1);
+          }}
+          className="flex h-full flex-col items-center justify-center rounded-l-lg px-4 hover:bg-gray-10"
+        >
+          -
+        </button>
+        <SeparatorVertical />
+        <input
+          type="text"
+          className="flex w-16 items-center justify-center !rounded-none border-0 text-center !outline-none !ring-0"
+          maxLength={3}
+          value={totalUsers}
+          onChange={(e) => {
+            e.preventDefault();
+            setTotalUsers(Number(e.target.value));
+          }}
+          onKeyDown={(e) => {
+            e.stopPropagation();
+            setShowButton(true);
+            if (e.key === 'Enter') {
+              e.preventDefault();
+              onInputChanges();
+            }
+          }}
+        />
+        <SeparatorVertical />
+        <button
+          disabled={disablePlusButton}
+          onClick={(e) => {
+            e.preventDefault();
+            onSeatsChange(seats + 1);
+            setTotalUsers(seats + 1);
+          }}
+          className="flex h-full flex-col items-center justify-center rounded-r-lg px-4 hover:bg-gray-10"
+        >
+          +
+        </button>
+      </div>
+      {showButton ? (
```

Comment:
I think it would be better to change the name of the conditional constant to something not so generic, like: showApplyButton, or something that identifies which button is going to be shown